src/runtime_src/driver/xclng/xrt/user_gem/shim.cpp

```diff
@@ -1841,18 +1845,62 @@ uint xocl::XOCLShim::xclGetNumLiveProcesses()
   if(dev->user) {
     std::vector<std::string> stringVec;
     dev->user->sysfs_get("", "kdsstat", errmsg, stringVec);
-    // Dependent on message format built in kdsstat_show. Checking number of "context" in kdsstat.
-    // kdsstat has "context: <number_of_live_processes>"
+
+    /* Dependent on message format built in kdsstat_show. Checking number of "context" and "client pid" in kdsstat.
+     * kdsstat has
+       "context: <number_of_live_processes>
+        client pid:
+                    <pid> 
+                    <pid>
+       "
+     */ 
     if(stringVec.size() >= 4) {
```

Comment:
This is much too complicated parsing which will be close to impossible to get right when/if we change the format of kdsstat.  Please use std::regex for exacting what you need.